### PR TITLE
chore(syncing): we can only set peer to nullptr if both syncing flags…

### PR DESCRIPTION
**Pull request by MatusKysel**
… are false

### CODE REVIEW

File: libraries/core_libs/network/src/tarcap/shared_states/syncing_state.cpp [hunk 1]
```diff
@@ -42,14 +43,19 @@ void SyncingState::set_pbft_syncing(bool syncing, uint64_t current_period, std::
...
 
-void SyncingState::set_dag_syncing(bool syncing, std::shared_ptr<TaraxaPeer> peer) {
+void SyncingState::set_dag_syncing(bool syncing, std::shared_ptr<TaraxaPeer> peer /*=nullptr*/) {
   assert((syncing && peer) || !syncing);
   dag_syncing_ = syncing;
-  set_peer(std::move(peer));
+  // We set a peer if peer is not null and we are syncing or
+  // we set peer to null if dag and pbft syncing are not enabled
+  if ((syncing && peer) || (!syncing && !pbft_syncing_)) {
+    set_peer(std::move(peer));
```
Comment (mfrankovi) [inline libraries/core_libs/network/src/tarcap/shared_states/syncing_state.cpp:57]: I think we are basically checking the same thing in two places in set_pbft_syncing and set_dag_syncing, maybe on both places we only call set_peer and then inside of set_peer we do the check. But I am fine as is as well

**Review by mfrankovi — APPROVED** [review]

File: libraries/core_libs/network/src/tarcap/shared_states/syncing_state.cpp (hunk 1 above)
Comment (MatusKysel) [reply libraries/core_libs/network/src/tarcap/shared_states/syncing_state.cpp:57]: you are right, that is better

File: libraries/core_libs/network/src/tarcap/shared_states/syncing_state.cpp [hunk 2]
```diff
@@ -10,6 +10,10 @@ SyncingState::SyncingState(uint16_t deep_syncing_threshold)
 
 void SyncingState::set_peer(std::shared_ptr<TaraxaPeer> &&peer) {
   std::unique_lock lock(peer_mutex_);
+  // we set peer to null if dag and pbft syncing are not enabled
+  if (!peer && (pbft_syncing_ || dag_syncing_)) {
```
Comment (kstdl) [inline libraries/core_libs/network/src/tarcap/shared_states/syncing_state.cpp:14]: I think it is the tough part of code to read. Could we explicit set it to null at least?

Comment (kstdl) [reply libraries/core_libs/network/src/tarcap/shared_states/syncing_state.cpp:14]: I guess it is just easier to read then that one. But I am not sure if there could be a situation that value are other then null value here. 
```
if (!pbft_syncing_ && !dag_syncing) {
  peer_ = nullptr;
}
```

**Review by kstdl — APPROVED** [review]

**Pull request merged**